File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff_dict = {}
        count = 0
        for num in nums:
            diff_val = target - num
            if num in diff_dict:
                return [diff_dict.get(num), count]
            else:
                diff_dict[diff_val] = count
            count = count + 1
        return []

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const diffDict = {};
    let count = 0;
    for (let num of nums){
        let diff_val = target - num;
        if (num in diffDict)
            return [diffDict[num], count];
        else
            diffDict[diff_val] = count;
        count++;
    }
    return []
};